packages/@aws-cdk/aws-redshift-alpha/lib/private/privileges.ts

```diff
@@ -62,33 +62,25 @@ export class UserTablePrivileges extends Construct {
         username: props.user.username,
         tablePrivileges: cdk.Lazy.any({
           produce: () => {
-            const reducedPrivileges = this.privileges.reduce((privileges, { table, actions }) => {
-              const tableId = table.node.id;
-              if (!(tableId in privileges)) {
-                privileges[tableId] = {
+            const groupedPrivileges = this.privileges.reduce(
+              (privileges, { table, actions }) => ({
+                ...privileges,
+                [table.node.id]: {
+                  actions: [
+                    ...(privileges[table.node.id]?.actions ?? []),
+                    ...actions,
+                  ],
                   tableName: table.tableName,
-                  actions: [],
-                };
-              }
-              actions = actions.concat(privileges[tableId].actions);
-              if (actions.includes(TableAction.ALL)) {
-                actions = [TableAction.ALL];
-              }
-              if (actions.includes(TableAction.UPDATE) || actions.includes(TableAction.DELETE)) {
-                actions.push(TableAction.SELECT);
-              }
-              privileges[tableId] = {
-                tableName: table.tableName,
-                actions: Array.from(new Set(actions)),
-              };
-              return privileges;
-            }, {} as { [key: string]: { tableName: string; actions: TableAction[] } });
-            const serializedPrivileges: SerializedTablePrivilege[] = Object.entries(reducedPrivileges).map(([tableId, config]) => ({
+                },
+              }),
+              {} as Record<string, { tableName: string; actions: TableAction[] }>,
+            );
```

Comment:
Can we make this code more readable? It's already inside `reduce` and the use of spread operator with arrow function make this code hard to read. IMO, this can be simple and straightforward like
```
const groupedPrivileges = this.privileges.reduce(
  (acc, { table, actions }) => {
    const currentActions = acc[table.node.id]?.actions ?? [];
    const updatedActions = [...currentActions, ...actions];

    acc[table.node.id] = {
      actions: updatedActions,
      tableName: table.tableName,
    };

    return acc;
  },
  {} as Record<string, { tableName: string; actions: TableAction[] }>
);
```